### ISSUE TITLE
CA3147 does not account for async 

### ISSUE THREAD

**Issue by chinwobble** [dotnet/roslyn-analyzers#2721]
#### Analyzer package

Microsoft.NetFramework.Analysers

#### Package Version

v2.9.4 (Latest)

#### Diagnostic ID
CA3147

#### Repro steps
the following method should give a warning for CA3147 but it doesn't since its async. 
```
[HttpPost]
public async Task<ActionResult> Post(PostModel model) 
{
// do stuff
}
```

#### Expected behavior
warning should be displayed for async methods

#### Actual behavior
warning only works for sync methods.

**Comment by mavasani** [comment]
Tag @dotpaul @LLLXXXCCC

**Comment by dotpaul** [comment]
Thanks @chinwobble !  I will look into this.
